sdk/openai/openai-assistants/review/openai-assistants.api.md

```diff
@@ -0,0 +1,644 @@
+    constructor(key: string);
+    get key(): string;
+    update(newKey: string): void;
+}
+
+// @public
+export interface RequiredAction {
+    submitToolOutputs?: SubmitToolOutputsDetails;
+    type: string;
+}
+
+// @public
+export interface RequiredFunctionToolCall {
+    function: FunctionDefinition;
+    id: string;
+    type: "function";
+}
+
+// @public
+export type RequiredToolCall = RequiredFunctionToolCall;
+
+// @public
+export interface RetrievalToolCall {
+    id: string;
+    retrieval: Record<string, string>;
+    type: "retrieval";
+}
+
+// @public
+export interface RetrievalToolDefinition {
+    type: "retrieval";
+}
+
+// @public
+export interface RunError {
+    code: string;
+    message: string;
+}
+
+// @public
+export interface RunError {
+    code: string;
+    message: string;
+}
+
+// @public
+export type RunStatus = string;
+
+// @public
+export interface RunStep {
+    assistantId: string;
+    cancelledAt: Date | null;
+    completedAt: Date | null;
+    createdAt: Date;
+    expiredAt: Date | null;
+    failedAt: Date | null;
+    id: string;
+    lastError: RunStepError | null;
+    metadata?: Record<string, string> | null;
+    runId: string;
+    status: RunStepStatus;
+    stepDetails: RunStepDetails;
+    threadId: string;
+    type: RunStepType;
+}
+
+// @public
+export type RunStepDetails = RunStepMessageCreationDetails | RunStepToolCallDetails;
+
+// @public
+export interface RunStepError {
+    code: RunStepErrorCode;
+    message: string;
+}
+
+// @public
+export interface RunStepError {
+    code: RunStepErrorCode;
+    message: string;
+}
+
+// @public
+export type RunStepErrorCode = string;
+
+// @public
+export interface RunStepMessageCreationDetails {
+    messageCreation: RunStepMessageCreationReference;
+    type: "message_creation";
+}
+
+// @public
+export interface RunStepMessageCreationReference {
+    messageId: string;
+}
+
+// @public
+export type RunStepStatus = string;
+
+// @public
+export interface RunStepToolCallDetails {
+    toolCalls: ToolCall[];
+    type: "tool_calls";
+}
+
+// @public
+export type RunStepType = string;
+
+// @public
+export interface SubmitToolOutputsDetails {
+    toolCalls: RequiredToolCall[];
+}
+
+// @public
+export interface SubmitToolOutputsToRunOptions extends OperationOptions {
+}
+
+// @public
+export interface ThreadDeletionStatus extends DeletionStatus {
+}
+
+// @public
+export interface ThreadMessage {
+    assistantId?: string;
+    content: MessageContent[];
+    createdAt?: Date;
+    fileIds?: string[];
+    id?: string;
+    metadata: Record<string, string> | null;
+    role: string;
+    runId?: string;
+    threadId?: string;
+}
+
+// @public
+export interface ThreadRun {
+    assistantId: string;
+    cancelledAt: Date | null;
+    completedAt: Date | null;
+    createdAt: Date;
+    expiresAt: Date | null;
+    failedAt: Date | null;
+    fileIds: string[];
+    id: string;
+    instructions: string;
```

Comment:
Is this always present?

Reply:
Yes, according to the generated code, instructions is always returned with the ThreadRun. TypeSpec [here](https://github.com/Azure/azure-rest-api-specs/blob/d541e6f0aee7a22c1f5c9d77635824c8d784a7ce/specification/ai/OpenAI.Assistants/runs/models.tsp#L46)